gcovr/formats/cobertura/read.py

```diff
@@ -0,0 +1,141 @@
+#
+def read_report(options: Options) -> CovData:
+    """merge a coverage from multiple reports in the format
+    compatible with Cobertura"""
+
+    if len(options.cobertura_add_tracefile) == 0:
+        return None
+
+    datafiles = set()
+
+    for trace_files_regex in options.cobertura_add_tracefile:
+        trace_files = glob(trace_files_regex, recursive=True)
+        if not trace_files:
+            raise RuntimeError(
+                "Bad --add-tracefile option.\n" "\tThe specified file does not exist."
+            )
+        else:
+            for trace_file in trace_files:
+                datafiles.add(os.path.normpath(trace_file))
+
+    covdata = dict()
+    for filename in datafiles:
+        LOGGER.debug(f"Processing XML file: {filename}")
+
+        try:
+            root = etree.parse(filename).getroot()
+        except Exception:
+            raise RuntimeError(
+                "Bad --add-tracefile option.\n"
+                f"\tThe file {filename} is not a valid XML file."
+            )
+
+        for gcovr_file in root.xpath("./packages//class"):
+            if gcovr_file.get("filename") is None:
+                LOGGER.warning(
+                    f"missing filename attribute in class element of {filename}"
+                )
+                continue
+
+            file_path = os.path.join(
+                os.path.abspath(options.root),
+                os.path.normpath(gcovr_file.get("filename")),
+            )
+
+            filtered, excluded = apply_filter_include_exclude(
+                file_path, options.filter, options.exclude
+            )
+
+            # Ignore if the filename does not match the filter
+            if filtered:
+                LOGGER.debug(f"  Filtering coverage data for file {file_path}")
+                continue
+
+            # Ignore if the filename matches the exclude pattern
+            if excluded:
+                LOGGER.debug(f"  Excluding coverage data for file {file_path}")
+                continue
+
+            file_coverage = FileCoverage(file_path)
+            merge_options = get_merge_mode_from_options(options)
+            for xml_line in gcovr_file.xpath("./lines//line"):
+                insert_line_coverage(file_coverage, _line_from_xml(filename, xml_line))
+
+            insert_file_coverage(covdata, file_coverage, merge_options)
+
+    return covdata
+
+
+def _line_from_xml(filename: str, xml_line) -> LineCoverage:
+    try:
+        lineno = int(xml_line.get("number"))
+        count = int(xml_line.get("hits"))
+        is_branch = xml_line.get("branch") == "true"
+        branch_msg = xml_line.get("condition-coverage")
+    except Exception:
+        LOGGER.warning(f"Invalid line information in file {filename}")
+
+    line = LineCoverage(lineno, count=count)
+
+    if is_branch and branch_msg is not None:
+        try:
+            [covered, total] = branch_msg[branch_msg.find("(") + 1 : -1].split("/")
```

Comment:
Better use
```suggestion
            [covered, total] = branch_msg[branch_msg.rfind("(") + 1 : -1].split("/")
```

Reply:
Are you suggesting this for effciency. It looks like `rfind` will have less work to do when 100% of the branches are covered. Otherwise `find` is as good or better in all other cases. Happy to change it, just curious about the reason.

Reply:
It's more for security. You need the content of the braced block at the end and with refund you're independent from the content before. With refund the content before can also contain a open brace without affecting the result.